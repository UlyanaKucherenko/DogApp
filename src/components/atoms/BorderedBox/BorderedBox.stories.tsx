import { Meta, Story } from '@storybook/react';

import { alpha } from '@mui/material';
import { createTheme } from '@mui/material/styles';
import BorderedBox from '.';

const theme = createTheme();

const withHoverEffect = Story => (
  <div
    onMouseEnter={() => Story({ showHovered: true })}
    onMouseLeave={() => Story({ showHovered: false })}
  >
    <Story showHovered={false} />
  </div>
);

export default {
  title: 'BorderedBox',
  component: BorderedBox,
  decorators: [withHoverEffect]
};

export const HoverWithBlackShadow = () => (
  <BorderedBox
    showHovered={false}
    boxShadow={alpha(theme.palette.common.white, 1)}
  >
    Box with BLACK shadow on hover
  </BorderedBox>
);

export const HoverWithMainShadow = () => (
  <BorderedBox showHovered boxShadow={alpha(theme.palette.primary.main, 1)}>
    Box with MAIN shadow on hover
  </BorderedBox>
);
