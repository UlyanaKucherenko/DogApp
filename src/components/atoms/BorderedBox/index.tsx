import { Box, BoxProps, alpha } from '@mui/material';
import { styled } from '@mui/material/styles';

interface BorderedBoxProps extends BoxProps {
  borderRadius?: number;
  showHovered?: boolean;
}

const BorderedBox = styled(Box, {
  shouldForwardProp: prop => prop !== 'showHovered'
})<BorderedBoxProps>(({ theme, borderRadius, showHovered }) => ({
  borderRadius: borderRadius || 20,
  boxShadow: `6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`,
  transition: 'box-shadow 0.3s ease-in-out',
  '&:hover': {
    boxShadow: showHovered
      ? `6px 6px 8px 0px ${alpha(theme.palette.primary.main, 1)}`
      : `6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`
  },
  '&:active': {
    boxShadow: showHovered
      ? `6px 6px 8px 0px ${alpha(theme.palette.primary.main, 1)}`
      : `6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`
  }
}));

export default BorderedBox;
