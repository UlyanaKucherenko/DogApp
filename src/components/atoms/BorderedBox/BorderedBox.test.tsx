import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/';
import { alpha } from '@mui/material';
import { createTheme } from '@mui/material/styles';
import BorderedBox from '.';

const theme = createTheme();

describe('BorderedBox', () => {
  it('should render without errors', () => {
    render(<BorderedBox data-testid="bordered-box" showHovered={false} />);
    expect(screen.getByTestId('bordered-box')).toBeInTheDocument();
    expect('bordered-box').toMatchSnapshot();
  });

  it('the default state should be a black shadow', () => {
    render(<BorderedBox data-testid="bordered-box" showHovered={false} />);
    const borderedBox = screen.getByTestId('bordered-box');
    expect(borderedBox).toHaveStyle(
      `box-shadow: 6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`
    );
    expect('bordered-box').toMatchSnapshot();
  });

  it('should have a black shadow when transmitting this props', () => {
    render(<BorderedBox data-testid="bordered-box" showHovered={false} />);
    const borderedBox = screen.getByTestId('bordered-box');
    userEvent.hover(borderedBox);
    expect(borderedBox).toHaveStyle(
      `box-shadow: 6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`
    );
    expect('bordered-box').toMatchSnapshot();
  });

  it('should have a main shadow when transmitting this props', () => {
    render(<BorderedBox data-testid="bordered-box" showHovered />);
    const borderedBox = screen.getByTestId('bordered-box');
    userEvent.hover(borderedBox);
    expect(borderedBox).toHaveStyle(
      `box-shadow: 6px 6px 8px 0px ${alpha(theme.palette.primary.main, 1)}`
    );
    expect('bordered-box').toMatchSnapshot();
  });
});
