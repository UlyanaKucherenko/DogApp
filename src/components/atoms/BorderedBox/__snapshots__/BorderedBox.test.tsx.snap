// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderedBox should have a black shadow when transmitting this props 1`] = `"bordered-box"`;

exports[`BorderedBox should have a main shadow when transmitting this props 1`] = `"bordered-box"`;

exports[`BorderedBox should render without errors 1`] = `"bordered-box"`;

exports[`BorderedBox the default state should be a black shadow 1`] = `"bordered-box"`;
